microsoft-search/pnp-modern-search-layouts
# Feature/events custom results template

### Description
- Added a new Event Card layout within the Results folder
- The Event Card layout uses the [pnp-documentcard](https://microsoft-search.github.io/pnp-modern-search/extensibility/web_components_list/#pnp-documentcard) structure to present items from an Events list in an appropriate form

### Type of Change

### How Has This Been Tested?
- The steps outlined in the layout guide have been followed within a separate test tenant to ensure it reproduces the same results
- Steps to test the layout:
     - Populate a SharePoint events list with some events including banner images and participants
     - Create a pnp search page
     - Change the search scope to ```{searchTerms} Path:{Site} contentclass:"STS_ListItem_Events"```
     - Add required selected properties:
             `BannerUrlOWSURLH`
             `EventsRollUpCategory`
             `EventsRollUpEndDate`
             `EventsRollUpStarteDate`
             `ListItemID`
             `Location`
             `ParticipantsPickerOWSUSER`
     - Update `Author` Slot to `ParticipantsPickerOWSUSER`
     - Apply the custom template code

### Steps to test additional features
- Multiple participants - _shows as +1 under the first participant name_
     - Map the  `ParticipantsPickerOWSUSER` to a RefinableString property
     - Add the RefinableString property to the select properties
     - Update the `Author` slot to the RefinableString
- Category refiner - _enables the category link to refine the search results_
     - Add a refiner to the search page
     - Add the `EventsRollUpCategory`

### Checklist
